amundsen_application/static/js/components/TableDetail/DetailList/DetailListItem/index.tsx

```diff
@@ -30,6 +47,12 @@ class DetailListItem extends React.Component<DetailListItemProps, DetailListItem
     };
   }
 
+  openRequest = () => {
+    if (!this.props.requestIsOpen) {
+      this.props.toggleRequest();
+    }
```

Comment:
This isn't necessarily a blocker but I would prefer that our components are "dumb" whenever possible. Instead of checking `state.notification.requestIsOpen` this component could blindly fire off the `toggleRequest` and the reducer/saga can decide what to do.

The `toggleRequest` can also pass in some value (what column initiated this request) to prefill some options. Also this action can be changed to `openDataRequestDialog` or something. You can also make a separate action to explicitly close the dialog.

Reply:
@danwom changed the reducer/saga to support OPEN and CLOSE actions, got rid of the toggle since it wasn't super necessary. I'll add the passing of values in a follow up PR